### DISCUSSION TITLE
Why full mastodon permissions

### DISCUSSION THREAD

**Discussion by thibaultmol** [Traewelling/traewelling#2768] [Questions / Support] [answered]
I was wondering why the platform requires the full access to your entire account when you connect it with mastodon? If all you wanted to do is push a check in to mastodon then it shouldn't need the full permissions though right? 
![Screenshot_20240714-125556](https://github.com/user-attachments/assets/46cb89ea-81b0-4455-9431-80c561b7c6ac)

**Comment by HerrLevin** [comment, marked as answer]
You're right. The problem is the following: Our old mastodon library had a bug and didn't allow anything except read/write scopes. It's technically possible now to change it, but we would have to rewrite a huge chunk of our login/social posting code to handle both "old" keys with their scopes and the new ones with reduced scopes. Also we would have to re-authenticate all users and about 500 instances. Which is a lot.

**Comment by HerrLevin** [comment]
I've put some information into #2666, since this is a known issue with third party mastodon-services and nobody didn't notice that it's also mastodon itself. 🫣
